### ISSUE TITLE
[ERR_UNSUPPORTED_DIR_IMPORT]: Directory import 'project/node_modules/htm/preact' is not supported resolving ES modules imported from

### ISSUE THREAD

**Issue by TimDaub** [developit/htm#191]
👋🏼 

```bash
node -v
v14.15.4
```

And my project is using ESM `import { html } from "htm/preact";`. When I try to run this, I get the following error:

```
Error [ERR_UNSUPPORTED_DIR_IMPORT]: Directory import '/Users/user/Projects/project/node_modules/htm/preact' is not supported resolving ES modules impor
ted from /Users/user/Projects/project/test/build_test.mjs
  Did you mean to import htm/preact/index.js?
```

I know that it I can resolve it myself by passing `--experimental-specifier-resolution=node` to node, but I'd rather not do that.

> Did you mean to import htm/preact/index.js?

This works as a way of resolve it too. But is it safe for the future?

**Comment by developit** [comment]
Hiya @TimDaub - looks like we need to add Package Exports to this module.

**Comment by developit** [comment]
related: #181

**Comment by rschristian** [comment]
Late to the party here, but looks like [package exports were added just about a year ago now](https://github.com/developit/htm/commit/0f20aa265efd3355329bc4af7d827fa768f87ca2#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519). Anything planned before a release, as I imagine `exports` will need a major? 

Don't mean to pester, but with Node 10 being officially EOL and `"type": "module"` being supported from v12+, I imagine we're going to see an uptick in ESM usage here. I'm starting to switch some of my things over and I've seen a few authors are switching over to ESM-only, so I imagine more people will start running into this shortly.

**Comment by developit** [comment]
This should be fixed as of [version 3.1.0](https://github.com/developit/htm/releases/tag/3.1.0).
